Post Reactions

AuraOfDivinity opened the pull request:
Fixes #542 

- [x] Implements animated Reactions below posts as requested.
- [x] Implements a dropdown on the top right corner of the Post card.

![Reactions](https://user-images.githubusercontent.com/31003923/89172963-1a8b2900-d5a1-11ea-9953-1b0a99f856c6.gif)

This is only the frontend implementation. Some changes to the backend might be required for other reactions (Heart, Happy and Donut) since only upvotes are supported at the moment

AuraOfDivinity commented:
@jaskiratsingh2000 A few questions regarding this on my end,

1. I think we should separate the user reactions to a particular post from other functionalities such as "Adding to favorite" and bookmarking a post. We could have these functionalities on the top right dropdown for instance. 

2. How will we displaying the reactions to a particular post? Maybe we could have a separate modal to display just like on Facebook? Something like the following image? 

![image](https://user-images.githubusercontent.com/31003923/89173597-24f9f280-d5a2-11ea-8e8f-76806f0cd10f.png)

jaskiratsingh2000 commented:
Thanks for taking this up. Well, I think we could do in a same way as it is
showed inside the facebook

On Mon, 3 Aug 2020, 3:58 pm Asel Peiris, <notifications@github.com> wrote:

> @jaskiratsingh2000 <https://github.com/jaskiratsingh2000> A few questions
> regarding this on my end,
>
>    1.
>
>    I think we should separate the user reactions to a particular post
>    from other functionalities such as "Adding to favorite" and bookmarking a
>    post. We could have these functionalities on the top right dropdown for
>    instance.
>    2.
>
>    How will we displaying the reactions to a particular post? Maybe we
>    could have a separate modal to display just like on Facebook? Something
>    like the following image?
>
> [image: image]
> <https://user-images.githubusercontent.com/31003923/89173597-24f9f280-d5a2-11ea-8e8f-76806f0cd10f.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/codeuino/social-platform-donut-frontend/pull/556#issuecomment-667945329>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACOBHK5HYJCLV7WRCKLHR7LR62GOHANCNFSM4PTGFYKQ>
> .
>

Rupeshiya commented on package.json:
```diff
@@ -17,6 +17,7 @@
     "chart.js": "^2.9.3",
     "dotenv": "^8.2.0",
     "env-cmd": "^10.1.0",
+    "framer-motion": "^2.3.0",
```
@AuraOfDivinity I am not happy with this npm package, as it has 8 dependencies, can't do it with css transitions??

Rupeshiya commented on src/user/dashboard/news-feed/popups/PostReactionsModal.js:
```diff
@@ -0,0 +1,154 @@
+  );
+};
+PostReactionModal.propTypes = {
+  onClick: PropTypes.func,
```
onClick is not passed.

Rupeshiya replied:
Also connect to state, you forgot to connect

Rupeshiya commented:
Also @AuraOfDivinity Please add **Pin icon,** already told earlier as well.

Rupeshiya commented on src/actions/postAction.js:
```diff
@@ -1,95 +1,101 @@
-    if(res.status === 200) {
-      dispatch(setRequestStatus(true))
-      console.log('all posts ', res.data.posts)
+    const res = await axios.get(
```
@AuraOfDivinity Please remove these unnecessary format changes, this is pretty annoying issue

Rupeshiya commented on src/actions/postAction.js:
```diff
@@ -1,95 +1,101 @@
-    if(res.status === 200){
-      dispatch(setRequestStatus(true))
-      console.log('fetching all pinned posts ', res.data.pinnedPost)
+    const res = await axios.get(
```
un-necessary changes

AuraOfDivinity commented:
Updated reactions.

- [x] Added pin post icon
- [x] Added PostReactions Modal
- [x] Added text based on the number of reactions at the bottom of the component.

![ReactionsUpdated](https://user-images.githubusercontent.com/31003923/89431180-34716b00-d75d-11ea-8ea6-cad2fdffd393.gif)
